### DISCUSSION TITLE
How about a loop/repeat option?

### DISCUSSION THREAD

**Discussion by drscottlobo** [danigb/smplr#24] [Ideas]
Would it make sense to have a loop or repeat option for a scheduled set of sounds? Might be especially useful for the drum machine or a metronome that repeats a short phrase.

**Comment by naushad-jamil** [comment]
@danigb @drscottlobo 

I am getting good luck so far with smplr library. I am trying to implement a repeat option for a set of notes. I am just calling the schedule notes function exactly the same way I am calling it the first time. The first time audio is coming out fine. But on subsequent repeats I see the highlighting and state changes are all working except no audio.

I am calling the schedule function on an event that is scheduled after playback of the first set of scheduled notes are finished.

What am I missing?

Is there a recommended way to implement repeat of a selection of scheduled notes?

**Reply by danigb** [reply]
Can you provide some code, please?

**Reply by naushad-jamil** [reply]
It was very late last night when I put together some code for repeat feature. And saw this bug. I was too early to assume it was the library issue and posted this comment and went to bed. All kinds of thoughts start to appear as my eyes are closed and one of them was the cause for the bug. I realized, to support my tied notes I was using look ahead a flag for each note, isProcessed. In first iteration all notes get processed. When that same buffer is sent again all notes are skipped. resetting the flag solved the problem.

Thank you very much for getting back.
